tests/unit/s2n_self_talk_nonblocking_test.c

```diff
@@ -260,13 +255,31 @@ int test_send(int use_tls13, int use_iov, int prefer_throughput)
         const int client_rc = !use_iov ? mock_client(&io_pair, blob.data, data_size) : mock_client_iov(&io_pair, iov, iov_size);
 
         EXPECT_SUCCESS(s2n_io_pair_close_one_end(&io_pair, S2N_CLIENT));
-        _exit(client_rc);
+        EXPECT_OK(cleanup_io_data(&iov, iov_size, &blob));
+        exit(client_rc);
+    }
+
+    DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+    DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
+    DEFER_CLEANUP(struct s2n_cert_chain_and_key *chain_and_key = s2n_cert_chain_and_key_new(), s2n_cert_chain_and_key_ptr_free);
+
+    EXPECT_SUCCESS(s2n_read_test_pem(S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN, cert_chain_pem, S2N_MAX_TEST_PEM_SIZE));
+    EXPECT_SUCCESS(s2n_read_test_pem(S2N_DEFAULT_ECDSA_TEST_PRIVATE_KEY, private_key_pem, S2N_MAX_TEST_PEM_SIZE));
+
+    EXPECT_SUCCESS(s2n_cert_chain_and_key_load_pem(chain_and_key, cert_chain_pem, private_key_pem));
```

Comment:
Wouldn't this be another opportunity to use s2n_test_cert_chain_and_key_new?

Reply:
You could and I tried that but then you lose the DEFER_CLEANUP and need to cleanup chain_and_key manually.

I made the call that it would be approx same lines of code difference and this was fine.